spring-integration-mqtt/src/main/java/org/springframework/integration/mqtt/outbound/Mqttv5PahoMessageHandler.java

```diff
@@ -237,10 +265,17 @@ protected void publish(String topic, Object mqttMessage, Message<?> message) {
 		Assert.isInstanceOf(MqttMessage.class, mqttMessage, "The 'mqttMessage' must be an instance of 'MqttMessage'");
 		long completionTimeout = getCompletionTimeout();
 		try {
-			if (!this.mqttClient.isConnected()) {
-				this.mqttClient.connect(this.connectionOptions).waitForCompletion(completionTimeout);
+			IMqttAsyncClient theClient;
+			if (getClientManager() != null) {
+				theClient = getClientManager().getClient();
+			}
+			else {
+				if (!this.mqttClient.isConnected()) {
+					this.mqttClient.connect(this.connectionOptions).waitForCompletion(completionTimeout);
```

Comment:
I think we have to try to reconnect over here independently of the client source.
Even if client is provided by the manager, it is still legit to call this `connect()` in case of lost connection in the client.

Reply:
I guess you are right. Won't be harmful - except that if client is being connected in background, `connect` method will throw an `MqttException`. But this is probably a rare case.